src/coreclr/jit/lower.cpp

```diff
@@ -3400,10 +3369,138 @@ GenTree* Lowering::LowerSelect(GenTreeConditional* select)
         }
     }
 
+#ifdef TARGET_XARCH
```

Comment:
X86 only - did you plan on doing this for the other architectures?

Reply:
Yes, in a follow-up. As you can imagine it will be more involved to do for the compare chain. We might also need a new node type due to the "flags" immediate that is part of `ccmp`.

Reply:
> Yes, in a follow-up. As you can imagine it will be more involved to do for the compare chains. We might also need a new node type due to the "flags" immediate that is part of `ccmp`.

Did you still plan on doing both `SELECTCC` for arm64 and compare chains to `CCMP` in a single PR?

I'm rebasing my AND chains, and it's going to conflict heavily if your doing compare chains soon. I'm happy to hold off again if it's all in your plan.


Also, for CCMP nodes the cflags add a little big of complexity. Consider:
```
cmp     w0, #0
ccmp    w1, #0, nzc, eq
cset    x0, gt
```
The cflags`nzc` is created from the `gt` (inverting it then turning into flags). So that means the `CCMP` node needs to hold both the condition taken from the previous compare (`eq`) and it's own condition now held by the cset (`gt`). Either `CCMP` would have to have two gtCondition fields, or we'd have to have `CCMP_GT` etc.

Reply:
> Did you still plan on doing both SELECTCC for arm64 and compare chains to CCMP in a single PR?

Hmm, I suppose it will be necessary to change both at the same time. Let me prioritize working on this to unblock you and to avoid unnecessarily churning things multiple times.

> The cflagsnzc is created from the gt (inverting it then turning into flags). So that means the CCMP node needs to hold both the condition taken from the previous compare (eq) and it's own condition now held by the cset (gt).

Right, I think we can add a `GenTreeCCMP` node that has the flags immediate too. Or we could encode it as part of `gtFlags`. Or we could do what you said and have separate node types.

Reply:
> Hmm, I suppose it will be necessary to change both at the same time. Let me prioritize working on this to unblock you and to avoid unnecessarily churning things multiple times.

Ta! 

> Right, I think we can add a `GenTreeCCMP` node that has the flags immediate too. Or we could encode it as part of `gtFlags`. Or we could do what you said and have separate node types.

`gtFlags` feels bad because most of the flag space is in use, but I'm happy with any of the above.